bin/compile

```diff
@@ -9,6 +9,18 @@ CACHE_DIR=$2
 BIN_PATH="$BUILD_DIR/bin"
 TMP_PATH="$BUILD_DIR/tmp"
 mkdir -p "$CACHE_DIR" "$BIN_PATH" "$TMP_PATH"
+
+function download_package {
+  URL=$1
+  PKG_NAME=$2
+  PKG=$3
+  if [ -f "$PKG" ]; then
+    echo "-----> Using ${PKG_NAME} package from cache"
+  else
+    echo "-----> Downloading ${PKG_NAME} package"
+    curl $URL -L -s -o $PKG
+  fi
```

Comment:
Minor: as a follow-up, maybe extract the downloading/caching logic into a shared function, since you're doing it twice?